wordpress-mobile/WordPress-Android
# New design reader bottom sheet - step 1

Fixes #11017 and partially implements #10887 

Tried to separate the modifications a bit to not make all in one God PR.

| Phone | ![image](https://user-images.githubusercontent.com/47797566/73405497-555f8c00-42f4-11ea-9587-3466dac06f14.png) | ![image](https://user-images.githubusercontent.com/47797566/73405541-70320080-42f4-11ea-8f6d-8f709256f64f.png) |
|---|---|---|
| Tablet | ![image](https://user-images.githubusercontent.com/47797566/73405632-b25b4200-42f4-11ea-8e77-b0cd7b7f3253.png) | ![image](https://user-images.githubusercontent.com/47797566/73405651-bb4c1380-42f4-11ea-920d-458d7fc2e7c3.png) |

## Notes
The feature is active in the `zalpha` flavor.

What's covered in this PR:
1. The subfilter (the one with the settings cog) uses the `Filter` call to action when no filter is applied. When a filter is applied an `x` is presented near the filter name so that the user can remove the filter tapping on it
2. Bottom sheet now has `SITES` and `TAGS` tabs and on the header part presents rounded corners and a central handle
3. The bottom sheet is half expanded when opened both in portrait and landscape mode. 

## To test
The following test cases test the list of items reported above (same numbering applies)

### Test case 1
- Follow some sites and tags if not already
- Check that when no site/tag is selected in the subfilter it presents the `Filter` call to action
- Select a site/tag from the bottom sheet and check the correct label is presented in the subfilter and the posts from that stream are loaded in the reader
- Remove the filter (pressing on the `x` near the filter label) and check that again the `Filter` label is presented

### Test case 2
- Follow some sites and tags if not already
- Open the bottom sheet and check it is tabbed like the images above
- Check that the tabs title shows the number of followed sites and tags
- Smoke test the two lists checking the correct filter is selected and correct posts are loaded

### Test case 3
- Open the bottom sheet (both landscape and portrait) and check it is half height of the screen
- Same if rotating the device while bottom sheet is opened
- The half height is also enforced when the bottom sheed is expanded to full height and then collapsed.

PR submission checklist:
note: not adding accessibility here since there will be a dedicated audit of the feature to close ticket #10962
